openmls/src/tree/secret_tree.rs

```diff
@@ -127,49 +127,33 @@ impl SecretTree {
         size: TreeSize,
         own_index: LeafNodeIndex,
     ) -> Self {
-        let mut leaf_nodes = std::iter::repeat_with(|| None)
-            .take(size.leaf_count() as usize)
-            .collect::<Vec<_>>();
-
-        let mut parent_nodes = std::iter::repeat_with(|| None)
-            .take(size.parent_count() as usize)
-            .collect::<Vec<_>>();
-
-        match root(size) {
-            TreeNodeIndex::Leaf(leaf_index) => {
-                leaf_nodes[leaf_index.usize()] = Some(SecretTreeNode {
-                    secret: encryption_secret.consume_secret(),
-                });
-            }
-            TreeNodeIndex::Parent(parent_index) => {
-                parent_nodes[parent_index.usize()] = Some(SecretTreeNode {
-                    secret: encryption_secret.consume_secret(),
-                });
-            }
-        }
-
-        let handshake_sender_ratchets = std::iter::repeat_with(|| Option::<SenderRatchet>::None)
-            .take(size.leaf_count() as usize)
-            .collect();
-
-        let application_sender_ratchets = std::iter::repeat_with(|| Option::<SenderRatchet>::None)
-            .take(size.leaf_count() as usize)
-            .collect();
-
-        log::trace!(
-            "Created secret tree with {} leaves and {} nodes.",
-            leaf_nodes.len(),
-            parent_nodes.len()
-        );
-
-        SecretTree {
+        let leaf_count = size.leaf_count() as usize;
+        let leaf_nodes = std::iter::repeat_with(|| None).take(leaf_count).collect();
```

Comment:
Why not just `vec![None; leaf_count]`? No strong opinion, but it would make the code a little easier to read.

Reply:
Because `vec![]` requires the element type to implement `Clone` (even for `None`), which we don't want for `EcryptionSecret`.